src/vec_bcd_ppc.h

```diff
@@ -3268,6 +4076,273 @@ vec_quantize0_Decimal128 (_Decimal128 val)
 #endif
 }
 
+/** \brief Vector Decimal Convert Binary Coded Decimal (BCD) digit
+ *  pairs from radix 100 binary integer bytes.
+ *
+ *  Convert 16 radix 100 digits to 32 BCD Format decimal digits.
+ *  Input is radix 100 digits as binary bytes in the range 0-99.
+ *  Each byte converted to the equivalent BCD digit pair in adjacent
+ *  nibbles.
+ *
+ *  This can be used as the last stage operation in wider binary to
+ *  decimal conversions.
+ *
+ *  \note the nibble high to low digit word is effectively big endian.
+ *  This matches the digit order precedence of Decimal Add/Subtract.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 24-34 | 1/cycle  |
+ *  |power9   | 27-37 | 1/cycle  |
+ *
+ *  @param vra a 128-bit vector treated as a vector unsigned char of
+ *  radix 100 digits.
+ *  @return 128-bit vector unsigned char of BCD nibble pairs in the
+ *  range 0-9.
+ */
+static inline vui8_t
+vec_rdxcf100b (vui8_t vra)
+{
+  vui8_t result;
+  vui8_t x6, c6, high_digit;
+  /* Compute the high digit correction factor. For binary 100s to BCD
+   * this is the radix 100 value divided by 10 times by the radix
+   * difference in binary.  For this stage we use 0x10 - 10 = 6.  */
+  high_digit = vra / 10;
+#if 0
+  c6 = vec_splat_u8 ((unsigned char) 0x06);
```

Comment:
you want to leave this code in?

Reply:
I think:
#if (__GNUC__ > 6)
  x6 = high_digit * 6;
  result = vra + x6;
#else
  c6 = vec_splat_u8 ((unsigned char) 0x06);
  x6 = vec_mulubm (high_digit, c6);
  /* Add the high digit correction bytes to the original
   * radix 100 bytes in binary. */
  result = vec_add (vra, x6);
#endif